### ISSUE TITLE
Error in reshape during conversion

### ISSUE THREAD

**Issue by shashanktyagi** [onnx/onnx-tensorrt#19]
While converting YOLO onnx model, I get the following error. Any ideas as to what could be the cause of this? 
Parsing model                                                                                                                                                                                                                                                                    
While parsing node number 463 [Reshape -> "830"]:                                                                                                                                                                                                                                 
ERROR: /home/shashank/onnx-tensorrt/builtin_op_importers.cpp:900 In function importReshape:                                                                                                                                                                                      
[8] Assertion failed: new_shape.nbDims == 3

**Comment by yinghai** [comment]
What's the output shape of the reshape node?

**Comment by shashanktyagi** [comment]
Here are all the operations on shapes:
(N, 75, 13, 13) is reshaped to (N, 75, 13*13)
(N, 75, 13*13) is transposed to (N, 13*13, 75)
(N, 13*13, 75) is reshaped to (N, 13*13*3, 25)

**Comment by yinghai** [comment]
Yes, right now, we only support reshape to 2D or 4D tensors including the batch size: https://github.com/onnx/onnx-tensorrt/blob/master/builtin_op_importers.cpp#L893-L908

This might be something we can improve...

**Comment by benbarsdell** [comment]
Yes unfortunately this is not currently supported. However, an upcoming update will add support for it using TensorRT 4.

**Comment by shashanktyagi** [comment]
I also figured that. I was wondering if you had something to circumvent this problem for now. 
When is the updated expected to be released?

**Comment by shashanktyagi** [comment]
I found a makeshift solution to reshape by adding an extra dimension. This fixed the previous error. However, now I run into the following error - 

[2018-06-12 18:43:45   ERROR] runtime.cpp (16) - Cuda Error in allocate: 2
[2018-06-12 18:43:45   ERROR] runtime.cpp (16) - Cuda Error in allocate: 2
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to create object
Aborted (core dumped)

It doesn't look like OOM error. I would appreciate any cue on what caused this.

**Comment by yinghai** [comment]
@shashanktyagi https://github.com/onnx/onnx-tensorrt/pull/21 is in merged. See if that unblocks you?

**Comment by BeHappy1** [comment]
Hi, I meet a seem problem. my matrix is (18,512), but when i reshape it to (2,9,512), it will report the following question.when i reshape to (2,9,1,512),it's still wrong.so can you give me some advice.

![2018-09-29 18-40-25](https://user-images.githubusercontent.com/26538531/46244810-378bc980-c417-11e8-9330-f508d9804590.png)

**Comment by vaishnavm217** [comment]
Hi,
I think my issue is similar to @shashanktyagi 's initial problem. I am converting S3FD model to tensorRT.
```
----------------------------------------------------------------
Input filename:   s3fd_1280_sq.onnx
ONNX IR version:  0.0.3
Opset version:    6
Producer name:    pytorch
Producer version: 0.3
Domain:
Model version:    0
Doc string:
----------------------------------------------------------------
Parsing model
While parsing node number 51 [Reshape -> "117"]:
ERROR: /home/graymatics/vaishnav-algo/onnx-tensorrt/builtin_op_importers.cpp:1383 In function importReshape:
[8] Assertion failed: get_shape_size(new_shape) == get_shape_size(weights.shape)
```
here is the verbrose [output](https://github.com/onnx/onnx-tensorrt/files/2561268/graph.txt)

**Comment by kevinch-nv** [comment]
Are any of you in this thread still having issues? Have you tried parsing your models with the latest TensorRT release (7.2)?

**Comment by kevinch-nv** [comment]
Closing due to inactivity - if you are still having issues with the latest version of onnx-tensorrt feel free to open a new issue.
